sherpa/astro/io/__init__.py

```diff
@@ -58,6 +58,30 @@
 io_opt = config.get('options', 'io_pkg')
 io_opt = str(io_opt).strip().lower()
 
+# Python 3 allows the following (although should move to the dictionary
+# style provided by the mapping protocal access.
+#
+# ogip_emin = config.get('ogip', 'minimum_energy', fallback='1.0e-10')
+if config.has_option('ogip', 'minimum_energy'):
```

Comment:
I must be missing something, because I can't find where this configuration option is used. I see it's used to set `data['ethresh']` in `read_arf` and `read_rmf` but then I can't find any usage of this.

Reply:
data is used to set up the keyword arguments when initializing the `DataARF` or `DataRMF` objects - e.g. lines 371 and 397 below which say

```
    return DataARF(filename, **data)
    return DataRMF(filename, **data)
```

The constructors for these classes use the `ethresh` argument to set up the minimum energy threshold, which eventually gets passed to the `validate_ogip_energy_ranges` routine you commented on above.

Reply:
Ah! Thanks! Hardly explicit, but that's not your fault :+1:

Reply:
Does this need commenting, or restucturing?

Reply:
No, actually I am not sure how I missed it to begin with. I'll blame python ;)

Reply:
Just to say it, this is an example of the issues I see in the backend implementation. The `get_arf_data` is almost identical in the two backends, and really what differs is just how the rows and columns are extracted from the file according to different APIs (the various `_try_*` functions). So, when you read this code, a lot of things and validations happen in different places. Some data is populated and to some extent validates in the so-called backend, now some data is validated/populated in the `io` module, while some other info is populated/validated in the constructor (`validate_ogip_energy_ranges`). For what is worth I now understand your original question about where the best place to put the validation is. Let me see if I can do something sensible to untangle this in the context of this PR.